sds_data_manager/lambda_code/IAlirtCode/ialirt_data_query_api.py

```diff
@@ -0,0 +1,245 @@
+        The error message.
+
+    Returns
+    -------
+    error : dict
+        The error dictionary.
+    """
+    return {
+        "statusCode": code,
+        "body": json.dumps({"message": message}),
+        "headers": {"Content-Type": "application/json"},
+    }
+
+
+def lambda_handler(event, context):
+    """Create metadata and add it to the database.
+
+    This function is an event handler for s3 ingest bucket.
+    It is also used to ingest data to the DynamoDB table.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    context : LambdaContext
+        This object provides methods and properties that provide
+        information about the invocation, function,
+        and runtime environment.
+
+    """
+    params = event.get("queryStringParameters") or {}
+
+    # --- Determine key condition ---
+    allowed_params = {
+        "instrument",
+        "time_utc_start",
+        "time_utc_end",
+        "met_in_utc_start",  # for backward compatibility
+        "met_in_utc_end",  # for backward compatibility
+    }
+
+    # Ensure allowed parameters
+    unexpected = set(params) - allowed_params
+    if unexpected:
+        return _error(400, f"Unexpected parameters: {', '.join(unexpected)}")
+
+    if not params.get("instrument"):
+        meta_instrument = "all"
+        meta_type = "science"
+    elif params["instrument"] == "spice":
+        meta_instrument = "spice"
+        meta_type = "spice"
+    elif params["instrument"].endswith("hk"):
+        meta_instrument = params["instrument"]
+        meta_type = "hk"
+    else:
+        meta_instrument = params["instrument"]
+        meta_type = "science"
+
+    # Get instrument or default to all.
+    requested_instrument = params.get("instrument")
+    instruments = (
+        [requested_instrument]
+        if requested_instrument
+        else ["hit", "mag", "codice_lo", "codice_hi", "swapi", "swe"]
+    )
+
+    items = []
+    query_time_total = 0
+
+    for instrument in instruments:
+        key_expr = Key("instrument").eq(instrument)
+        query_kwargs = {"KeyConditionExpression": key_expr}
+
+        if any(
+            param in params
+            for param in (
+                "time_utc_start",
+                "time_utc_end",
+                "met_in_utc_start",
+                "met_in_utc_end",
+            )
+        ):
+            result = apply_time_filters(params, query_kwargs)
+            if result is None:
+                return _error(400, "Invalid time format: expected YYYY-MM-DDTHH:MM:SS")
+            query_kwargs, range_start, range_end = result
+            # Checks if the max time range is exceeded.
+            if abs(
+                datetime.fromisoformat(range_end) - datetime.fromisoformat(range_start)
+            ) > timedelta(days=1):
+                return _error(400, "Start and end time cannot exceed 1 day apart.")
```

Comment:
Can you put this check in the `apply_time_filters` call as well so all time handling is in one location? You could maybe raise the `BadTimeException` with the message you want to deliver to the end user depending on the case that fails and then reformat that into the json response.